pkg/pfcpsim/errors.go

```diff
@@ -1,10 +1,40 @@
 package pfcpsim
 
-import "errors"
+import "fmt"
 
-// TODO create custom error types
+type pfcpSimError struct {
+	message string
+	error   error
+}
 
-var InvalidCauseErr = errors.New("InvalidCause")
-var TimeoutOccurredErr = errors.New("TimeoutOccurred")
-var InvalidResponseErr = errors.New("InvalidResponse")
-var AssociationInactiveErr = errors.New("AssociationInactive")
+func (e *pfcpSimError) Error() string {
+	return fmt.Sprintf("Message: %v, Error: %v", e.message, e.error)
+}
+
+func NewInvalidCauseError(err error) *pfcpSimError {
+	return &pfcpSimError{
+		message: "Invalid Cause from response",
+		error:   err,
+	}
+}
+
+func NewAssociationInactiveError(err error) *pfcpSimError {
+	return &pfcpSimError{
+		message: "Could not complete operation. Association is not active",
+		error:   err,
+	}
+}
+
+func NewTimeoutExpiredError(err error) *pfcpSimError {
+	return &pfcpSimError{
+		message: "Timeout has expired",
+		error:   err,
+	}
+}
+
+func NewInvalidResponseError(err error) *pfcpSimError {
```

Comment:
Can you try to use variadic arguments, so that there will be no need to pass `nil` argument if we don't have a child error?

Reply:
good idea!